src/monodroid/jni/monodroid-glue.cc

```diff
@@ -997,28 +997,30 @@ create_domain (JNIEnv *env, jclass runtimeClass, jstring_array_wrapper &runtimeA
 		free (domain_name);
 	}
 
-	if (is_running_on_desktop && is_root_domain) {
-		// Check that our corlib is coherent with the version of Mono we loaded otherwise
-		// tell the IDE that the project likely need to be recompiled.
-		char* corlib_error_message = const_cast<char*>(mono_check_corlib_version ());
-		if (corlib_error_message == nullptr) {
-			if (!androidSystem.monodroid_get_system_property ("xamarin.studio.fakefaultycorliberrormessage", &corlib_error_message)) {
+	if constexpr (is_running_on_desktop) {
+		if (is_root_domain) {
+			// Check that our corlib is coherent with the version of Mono we loaded otherwise
+			// tell the IDE that the project likely need to be recompiled.
+			char* corlib_error_message = const_cast<char*>(mono_check_corlib_version ());
+			if (corlib_error_message == nullptr) {
+				if (!androidSystem.monodroid_get_system_property ("xamarin.studio.fakefaultycorliberrormessage", &corlib_error_message)) {
+					free (corlib_error_message);
+					corlib_error_message = nullptr;
+				}
+			}
+			if (corlib_error_message != nullptr) {
+				jclass ex_klass = env->FindClass ("mono/android/MonoRuntimeException");
```

Comment:
If things are *really* FUBAR, this type won't exist.  Should we instead throw `java.lang.Error`, which should (!) *always* exist?

Reply:
....but that would be a semantic break from existing code.  Never mind.